packages/plugins/i18n/admin/src/components/LocaleSelect/index.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { Select, Option } from '@strapi/parts/Select';
+import { useIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import useLocales from '../../hooks/useLocales';
+import useDefaultLocales from '../../hooks/useDefaultLocales';
+import { getTrad } from '../../utils';
+
+/**
+ * The component is memoized and needs a useCallback over the onLocaleChange and
+ * onClear props to prevent the Select from re-rendering N times when typing on a specific
+ * key in a formik form
+ */
+const LocaleSelect = React.memo(({ value, onLocaleChange, error, onClear }) => {
+  const { formatMessage } = useIntl();
+  const { defaultLocales, isLoading } = useDefaultLocales();
+  const { locales } = useLocales();
+
+  const options = (defaultLocales || [])
+    .map(locale => ({
+      label: locale.code,
+      value: locale.name,
+    }))
+    .filter(({ label }) => {
+      const foundLocale = locales.find(({ code }) => code === label);
+
+      return !foundLocale;
+    });
+
+  const defaultOption = options[0];
+
+  return (
+    <Select
+      label={formatMessage({
+        id: getTrad('Settings.locales.modal.locales.label'),
+      })}
+      onClear={value ? onClear : undefined}
+      clearLabel={formatMessage({
+        id: getTrad('Settings.locales.modal.base.clearLabel'),
+      })}
+      error={error}
+      value={value || defaultOption}
+      onChange={selectedLocaleKey => {
+        const selectedLocale = options.find(locale => locale.label === selectedLocaleKey);
+
+        if (selectedLocale) {
+          onLocaleChange({ code: selectedLocale.label, displayName: selectedLocale.value });
```

Comment:
This needs to be switch to `{ code: selectedLocale.value, displayName: selectedLocale.label }`